include/seqan3/io/alignment_file/input.hpp

```diff
@@ -0,0 +1,1323 @@
+
+        // possibly add intermediate compression stream
+        secondary_stream = detail::make_secondary_istream(*primary_stream);
+
+        // initialize reference information
+        set_references(ref_ids, ref_sequences);
+
+        // buffer first record
+        read_next_record();
+    }
+
+    //!\overload
+    template <IStream2 stream_t,
+              std::ranges::ForwardRange ref_ids_t,
+              std::ranges::ForwardRange ref_sequences_t,
+              AlignmentFileInputFormat file_format>
+        requires std::Same<value_type_t<ref_sequences_t>, ref_sequence_type> &&
+                 std::Same<value_type_t<ref_ids_t>, ref_id_type>
+    alignment_file_input(stream_t                && stream,
+                         ref_ids_t const & ref_ids,
+                         ref_sequences_t const & ref_sequences,
+                         file_format        const & SEQAN3_DOXYGEN_ONLY(format_tag),
+                         selected_field_ids const & SEQAN3_DOXYGEN_ONLY(fields_tag) = selected_field_ids{}) :
+        primary_stream{new stream_t{std::move(stream)}, stream_deleter_default},
+        format{file_format{}}
+    {
+        static_assert(meta::in<valid_formats, file_format>::value,
+                      "You selected a format that is not in the valid_formats of this file.");
+
+        // possibly add intermediate compression stream
+        secondary_stream = detail::make_secondary_istream(*primary_stream);
+
+        // initialize reference information
+        set_references(ref_ids, ref_sequences);
+
+        // buffer first record
+        read_next_record();
+    }
+    //!\}
+
+    /*!\name Range interface
+     * \brief Provides functions for record based reading of the file.
+     * \{
+     */
+    /*!\brief Returns an iterator to current position in the file.
+     * \returns An iterator pointing to the current position in the file.
+     *
+     * Equals end() if the file is at end.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator begin() noexcept
+    {
+        return {*this};
+    }
+
+    /*!\brief Returns a sentinel for comparison with iterator.
+     * \returns Iterator to the first element.
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    sentinel end() noexcept
+    {
+        return {};
+    }
+
+    /*!\brief Return the record we are currently at in the file.
+     * \returns A reference to the currently buffered record.
+     *
+     * This function returns a reference to the currently buffered record, it is identical to dereferencing begin(),
+     * but begin also always points to the current record on single pass input ranges:
```

Comment:
```suggestion
     * and begin also always points to the current record on single pass input ranges:
```

?
but doesn't seem to fit here